### PR TITLE
Pre-app-router se-2 backmerge

### PR DESCRIPTION
**Pull request by rin-st**
Needed for [event indexes](https://github.com/scaffold-eth/se-2-challenges/issues/95).
Also, I think this pr will help to move to the app router later.

Copy-pasted se-2 code, removed themes and most of text-color changes

### CODE REVIEW

File: packages/nextjs/components/scaffold-eth/Input/InputBase.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const InputBase = <T extends { toString: () => string } | undefined = str
     <div className={`flex border-2 border-base-300 bg-base-200 rounded-full text-accent ${modifier}`}>
       {prefix}
       <input
-        className="input input-ghost focus:outline-none focus:bg-transparent focus:text-secondary-content h-[2.2rem] min-h-[2.2rem] px-4 border w-full font-medium placeholder:text-accent/50 text-secondary-content"
+        className="input input-ghost focus:outline-none focus:bg-transparent focus:text-gray-400 h-[2.2rem] min-h-[2.2rem] px-4 border w-full font-medium placeholder:text-accent/50 text-gray-400"
```
Comment (technophile-04) [inline packages/nextjs/components/scaffold-eth/Input/InputBase.tsx:38]: I think should put it back to `text-secondary-content`  🙌 : 

with `text-gray-400` : 
Screenshot 2023-09-30 at 8 35 03 PM

File: packages/nextjs/components/scaffold-eth/Contract/DisplayVariable.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const DisplayVariable = ({ contractAddress, abiFunction, refreshDisplayVa
           )}
         </button>
       </div>
-      <div className="text-secondary-content font-medium flex flex-col items-start">
+      <div className="text-gray-500 font-medium flex flex-col items-start">
```
Comment (technophile-04) [inline packages/nextjs/components/scaffold-eth/Contract/DisplayVariable.tsx:48]: same here as https://github.com/scaffold-eth/se-2-challenges/pull/96/files#r1341972348

File: packages/nextjs/next.config.js [hunk 3]
```diff
@@ -3,12 +3,15 @@
...
   reactStrictMode: true,
-  // Ignoring typescript/eslint errors during build (deploy won't fail even if there are errors)
   typescript: {
-    ignoreBuildErrors: true,
+    ignoreBuildErrors: process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true",
   },
   eslint: {
-    ignoreDuringBuilds: true,
+    ignoreDuringBuilds: process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true",
+  },
```
Comment (technophile-04) [inline packages/nextjs/next.config.js:11]: Lets put back to `true` by default 🙌

**Comment by technophile-04** [comment]
Tysm Rinat 🙏 looking great! added a couple of comments which are specific changes to the SE-2-chall

**Comment by rin-st** [comment]
Thank you! Updated pr.

Also locked ts version to 5.1.6 since after 5.2 I have errors in all the files
Снимок экрана 2023-09-30 в 20 13 47

We don't have that error in se-2 because of lock file. Let's fix it after app-routes pr will be merged. I think when locking all the packages.

See also https://stackoverflow.com/questions/76996326/parsing-error-deprecationerror-originalkeywordkind-has-been-deprecated-since

**Review by Pabl0cks — APPROVED** [review]
Awesome job Rinat! 🙌

Been playing with it and it's looking great to me 👌

**Comment by technophile-04** [comment]
Merging this 🙌 Tysm Rinat again !

**Comment by rin-st** [comment]
Oh I didn't notice second className in input-base, line is too long 😄 . Thank you for your thoroughness!

**Pull request merged**